docs/transformations/Anonymization.md

```diff
@@ -0,0 +1,41 @@
+# Java/Kotlin Anonymization: common part
+
+## Overview
+
+This transformation renames most named entities in a Java/Kotlin program to have
+automatically generated names.
+
+## Naming scheme
+
+Each named entity is renamed to have the form
+`parentName + "_" + kindPrefix + kindIndex`, where:
+
+- `parentName` is the name of the entity's parent after renaming. 
+  The parent is usually the
+  most nested scope owner that the entity's definition resides in.
+  For example, the parent of a class method is the corresponding class.
+  Global entities do not have a parent and, for them, `parentName` is
+  considered to be empty. In this case, the separator `_` is also omitted.
+  
+- `kindPrefix` is one of the following, depending on what kind of entity
+  is being renamed:
+
+    | Prefix | Kind                                          |
+    | ------ | ----                                          |
+    | `f`    | Function                                      |
+    | `sf`   | Static (or Companion Object) function         |
+    | `v`    | Variable                                      |
+    | `c`    | Class                                         |
+    | `init` | Constructor                                   |
```

Comment:
Sorry, I don't get it, why do we want to rename the constructor? Shouldn't it be the same as the class name?...